dialogue-apache-hc5-client/src/main/java/com/palantir/dialogue/hc5/ApacheHttpClientBlockingChannel.java

```diff
@@ -177,24 +176,7 @@ private static boolean requiresEmptyBody(Endpoint endpoint) {
     }
 
     private static void setBody(ClassicRequestBuilder builder, RequestBody body) {
-        builder.setEntity(new RequestBodyEntity(body, contentLength(builder)));
-    }
-
-    private static OptionalLong contentLength(ClassicRequestBuilder builder) {
-        Header contentLengthHeader = builder.getFirstHeader(HttpHeaders.CONTENT_LENGTH);
-        if (contentLengthHeader != null) {
-            builder.removeHeaders(HttpHeaders.CONTENT_LENGTH);
-            String contentLengthValue = contentLengthHeader.getValue();
-            try {
-                return OptionalLong.of(Long.parseLong(contentLengthValue));
-            } catch (NumberFormatException nfe) {
-                log.warn(
-                        "Failed to parse content-length value '{}'",
-                        SafeArg.of(HttpHeaders.CONTENT_LENGTH, contentLengthValue),
-                        nfe);
-            }
-        }
```

Comment:
I think the feign shim relies on this logic, we should support Content-Length in both places at least for a few releases

Reply:
Ack, I shall split a separate PR as discussed.